### PR TITLE
FE-1344 - track onboarding in segment

### PR DESCRIPTION
**Pull request by nstanard**
### What Changed
- Tracking onboarding steps in segment as long as it's not set to "done"

### How To Test
- Create a new account and go to the dashboard. 
- Go through each onboarding step manually calling the API to get past the sending domain verification process.
- Make sure calls are going out to segment for each step.

### To Do
- [x] Test
- [x] Feedback

### CODE REVIEW

File: src/helpers/segment.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ export const SEGMENT_EVENTS = {
   SENDING_DOMAIN_ADDED: 'Sending Domain Added',
   SENDING_DOMAIN_VERIFIED: 'Sending Domain Verified',
   EMPTY_STATE_LOADED: 'Empty state loaded',
+  ONBOARDING: 'Onboarding',
```
Comment (nstanard) [inline src/helpers/segment.js:22]: Not sure if this is what we want, maybe DashboardV2 Onboarding?

Comment (jonambas) [reply src/helpers/segment.js:22]: I would use a verb for the event so its consistent with the others, like "Dashboard Onboarding Loaded" or something

Comment (nstanard) [reply src/helpers/segment.js]: I like that. Updated!

File: src/pages/dashboardV2/DashboardPageV2.container.js [hunk 2]
```diff
@@ -66,6 +67,12 @@ function mapStateToProps(state) {
     onboarding = 'analyticsReportPromo';
   }
 
+  if (onboarding !== 'done') {
+    segmentTrack(SEGMENT_EVENTS.DASHBOARD_ONBOARDING, {
```
Comment (bkemper) [inline src/pages/dashboardV2/DashboardPageV2.container.js:71]: The desired result is to only call `segmentTrack` when `onboarding` has changed and doesn't each "done".  `mapStateToProps` on the other hand is going to be called whenever state changes which is more than we want.  I recommend moving this to a `useEffect` in the component to control number of times it is called.

Comment (nstanard) [reply src/pages/dashboardV2/DashboardPageV2.container.js]: testing now to confirm it's what we want

**Review by jonambas — APPROVED** [review]
This looks good to me – verified the track event is called once on load with the step

**Pull request merged**